src/Client.ts

```diff
@@ -108,12 +115,22 @@ export class Client {
             data,
         ]);
 
-        return this.requestRunner.run(<IRequestOptions>req)
-        .catch(err => {
+        return this.requestRunner.run(<IRequestOptions>req).catch(err => {
             if (this.provider) {
                 return this.provider.handleResponseError(err, <IRequestOptions>req);
             }
             throw err;
         });
     }
+
+    public createChatSocket(
+        ws: IGenericWebSocket,
+        endpoints: string[],
+        options: ISocketOptions,
+    ): Socket {
+        return new Socket(ws, endpoints, {
+            clientId: this.provider instanceof OAuthProvider ? this.provider.getClientId() : '',
```

Comment:
This should probably be null/undefined if we don't want clientId to show up in the headers/qs